src/Components/Components/src/Rendering/RenderTreeUpdater.cs

```
@@ -0,0 +1,113 @@
+                            {
+                                UpdateFrameToMatchClientState(
+                                    renderTreeBuilder,
+                                    framesArray,
+                                    closestElementFrameIndex,
+                                    frame.AttributeEventUpdatesAttributeName,
+                                    newFieldValue);
+                            }
+
+                            // Whether or not we did update the frame, that was the one that matches
+                            // the event handler ID, so no need to look any further
+                            return;
+                        }
+                        break;
+                }
+            }
+        }
+
+        private static void UpdateFrameToMatchClientState(RenderTreeBuilder renderTreeBuilder, RenderTreeFrame[] framesArray, int elementFrameIndex, string attributeName, object attributeValue)
+        {
+            // Find the attribute frame
+            ref var elementFrame = ref framesArray[elementFrameIndex];
+            var elementSubtreeEndIndexExcl = elementFrameIndex + elementFrame.ElementSubtreeLength;
+            for (var attributeFrameIndex = elementFrameIndex + 1; attributeFrameIndex < elementSubtreeEndIndexExcl; attributeFrameIndex++)
+            {
+                ref var attributeFrame = ref framesArray[attributeFrameIndex];
+                if (attributeFrame.FrameType != RenderTreeFrameType.Attribute)
+                {
+                    // We're now looking at the descendants not attributes, so the search is over
+                    break;
+                }
+
+                if (attributeFrame.AttributeName == attributeName)
+                {
+                    // Found an existing attribute we can update
+                    attributeFrame = attributeFrame.WithAttributeValue(attributeValue);
+                    return;
+                }
+            }
```

Is it ok to silently ignore the update if we can't find the attribute?